source/common/http/header_utility.cc

```diff
@@ -333,5 +333,57 @@ bool HeaderUtility::isModifiableHeader(absl::string_view header) {
           !absl::EqualsIgnoreCase(header, Headers::get().HostLegacy.get()));
 }
 
+HeaderUtility::HeaderValidationResult HeaderUtility::checkHeaderNameForUnderscores(
```

Comment:
It kills me that this code is now copied into a third place when we already effectively have it in both the H1 and H2 codec. This is really scary and security sensitive stuff and I feel we shouldn't be making the problem worse (it's in the opposite direction of https://github.com/envoyproxy/envoy/issues/10646). Is there any work we can do to share some of this code now? Maybe we can do a different PR to add this helper and call it somehow from both other codecs?

Reply:
I don't want to touch any code in production in this PR because that needs feature protection. Once this PR lands, the other two codec can switch to this utility function.

Reply:
OK fair enough. Can you at least add comments and put TODOs in the other codec places which reference this code and the tracking issue for merging?

Reply:
cc @yanavlasov might be a good starter project for one of the envoy-sec folks